superset/datasets/commands/samples.py

```diff
@@ -78,3 +116,24 @@ def validate(self) -> None:
             security_manager.raise_for_ownership(self._model)
         except SupersetSecurityException as ex:
             raise DatasetForbiddenError() from ex
+
+    @staticmethod
+    def get_limit_clause(
+        page: Optional[int], per_page: Optional[int]
+    ) -> Dict[str, int]:
+        samples_row_limit = app.config.get("SAMPLES_ROW_LIMIT", 1000)
```

Comment:
Isn't the global samples row limit already applied at [query_object_factory.py#L64](https://github.com/apache/superset/blob/d1861c7d3c95e72336466cde17fee9ae1be9155a/superset/common/query_object_factory.py#L64)?  In either case, wouldn't we want to ignore this limit for this purpose, in case they've set it to something really low?

Reply:
we cannot ignore the limit value in any detailed query(e.g. `select * from tbl`) because the unlimited query is a heavy load for some db.

Reply:
I meant that if the user has configured `SAMPLES_ROW_LIMIT` to be something really low, won't this not return full sets of results?  The 1000 limit fallback makes sense in case the API consumer requests crazy high page sizes but I'm worried that if the user has set `SAMPLES_ROW_LIMIT` to be something like 5, this will unexpectedly return pages of 5 results.

Reply:
That's what this setting does. for example, the SAMPLES_ROW_LIMIT = 10, but the source of the table has 100 rows, the samples will return 10 records, but the query of count(*)  will return a total count is 100, so the total_count is full count for this dataset(or query), and size of result always following the `SAMPLES_ROW_LIMIT` setting.

![image](https://user-images.githubusercontent.com/2016594/179769804-4f3a2f3d-a9f4-4828-b6f8-24be1c5c76e7.png)

Reply:
Right, and I was suggesting that we ignore `SAMPLES_ROW_LIMIT` for paginated results so the pages are always the requested size regardless of the `SAMPLES_ROW_LIMIT` setting (unless they request a very large page size), otherwise users might be surprised to find that their requested page size isn't being respected.

Reply:
Ok, understood.